Core/Lib/API/APIModel.php

```diff
@@ -185,28 +187,33 @@ public function doGET(): bool
     }
 
     /**
-     * Process the GET request. Overwrite this function to implement is functionality.
-     *
-     * @return bool
+     * Load the model and replace the past data in the loaded model.
+     * Returns true if the record already exists in the model, and false if not.
+     * If the id is passed as data and parameter, verify that both match, 
+     * returning error if there is inconsistency.
+     * 
+     * @return bool true if the resource exists in the model
      */
-    public function doPOST(): bool
+    private function getResource(): bool
     {
         $cod = $this->model->primaryColumn();
 
         // If editing, retrieve the current data
-        $this->model->loadFromCode($this->params[0]);
+        $exist = $this->model->loadFromCode($this->params[0]);
         $this->model->{$cod} = $this->params[0];
 
         // Retrieve the past data, and replace the changes
         $values = $this->request->request->all();
-        if (isset($values[$cod]) && $values[$cod] !== $this->params[0]) {
```

Comment:
you could get a invalid offset warning using params[0] no previously checked before, This could lead a potential bug too